File: eventos2/core/serializers/track_review_question.py
from rest_framework import serializers

from eventos2.core.models import Track, TrackReviewQuestion


class TrackReviewQuestionSerializer(serializers.ModelSerializer):
    class Meta:
        model = TrackReviewQuestion
        fields = ["id", "text", "answer_type"]

    def create(self, validated_data):  # pragma: no cover - no complexity
        raise NotImplementedError("Use TrackReviewQuestionCreateSerializer")


class TrackReviewQuestionCreateSerializer(serializers.ModelSerializer):
    track = serializers.PrimaryKeyRelatedField(queryset=Track.available_objects.all())

    class Meta:
        model = TrackReviewQuestion
        fields = ["track", "id", "text", "answer_type"]

File: frontend/api/trackReviewQuestion.js
export function createTrackReviewQuestionAPIClient($axios) {
  return {
    create({ trackId, text, answer_type }) {
      return $axios.$post(`api/v1/track_review_questions/`, {
        track: trackId,
        text,
        answer_type
      })
    }
  }
}
